pandas/__init__.py

```diff
@@ -110,6 +110,7 @@
 
 from pandas.core.dtypes.dtypes import SparseDtype
 
+from pandas import util
```

Comment:
For https://pandas.pydata.org/docs/reference/api/pandas.util.hash_pandas_object.html and other functions

Reply:
I think there are other methods in this module that are not meant to be public though: `capitalize_first_letter`, `cache_readonly`, `Substitution`, `Appender`

Reply:
Okay, I think we could do `__all__ = [all_classes_functions_in_the_docs]` to account for that. And/or we could deprecate it from utils and put it as part of a different namespace.

Reply:
`__all__` only changes behavior of `from foo import *`, no? What do we think of moving these functions (can we make a list of them?) to the top namespace?

Does this hold up other aspects of this PR? If not, I would suggest moving to a separate PR.

Reply:
Happy to remove this change from this PR. I will open an issue to first discuss how this should be handled.

`__all__` affects `from pandas.utils import *` but (more importantly) type checkers, use it (next to other rules) to determine which symbols are meant to be public: if `__all__` is present, only those symbols listed in it are considered public.

Reply:
> if __all__ is present, only those symbols listed in it are considered public.

This doesn't seem to be the case with mypy.

```
# foo.py
x = 5
y = 6

__all__ = ['y']

# bar.py
import foo

print(foo.x)
print(foo.y)
```

The result of `mypy bar.py`: `Success: no issues found in 1 source file`

Reply:
> if `__all__` is present, only those symbols listed in it are considered public.

This doesn't seem to be the case with mypy:

```
# test/__init__.py
x = 5
y = 6

__all__ = ['y']

# bar.py
import test

print(test.x)
print(test.y)
```

`mypy bar.py` reports no issues for me.

Reply:
I must have misremembered that (pyright also allows that).

Pyright describes how it handles it: https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface